src/new-components/avatar/avatar.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { render, screen } from '@testing-library/react'
+import React from 'react'
+
+import { Avatar } from './avatar'
+
+describe('Avatar', () => {
+    it('renders a background image when avatarUrl is supplied', () => {
+        render(getAvatar({ avatarUrl: 'https://foo.bar/com.png' }))
+
+        const avatar = screen.getByTestId('avatar')
+
+        expect(avatar).toMatchSnapshot()
+    })
+
+    it('renders initials of user name when avatarUrl is not supplied', () => {
+        render(getAvatar())
+
+        const avatar = screen.getByTestId('avatar')
+
+        expect(avatar).toMatchSnapshot()
+        expect(avatar.textContent).toBe('HM')
+    })
+
+    it('renders initials on custom color', () => {
+        render(getAvatar({ colorList: ['red', 'green', '#0000FF'] }))
+
+        const avatar = screen.getByTestId('avatar')
+
+        expect(avatar).toMatchSnapshot()
+        expect(avatar.textContent).toBe('HM')
+    })
+
+    it('renders initials of user email when avatarUrl is not supplied', () => {
+        render(getAvatar({ user: { email: 'henning@doist.com' } }))
+
+        const avatar = screen.getByTestId('avatar')
+
+        expect(avatar).toMatchSnapshot()
+        expect(avatar.textContent).toBe('H')
```

Comment:
Same observations about there being no need for the snapshot, and the use of `toHaveTextContent`.